### PR TITLE
Add create rule form (first pass)

### PR DESCRIPTION
**Pull request by rhystmills**
Co-authored-by @Fweddi 

## What does this change?

This PR adds a 'create rule' form. Currently the form is very permissive, and only the 'rule' is required.

image

The DB model for categories and and tags are subject to change (tags also currently aren't save to the database, PR-pending). In this PR, categories are a hardcoded list, in the future we'd want to use an endpoint, drawing from existing tags. We'd also want tags to be drawn from the database in the future, with users able to add their own tags.

The position of the form in this PR is subject to change - we've tried to cover the core functionality of the form (managing form data and submitting to the backend) as the UI will be finalised later.

## How to test

With Composer credentials, run `typerighter` locally using `./script/start`. Can you add a new rule via the form as intended?

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [X] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

**Review by jonathonherbert — COMMENTED** [review]
Looks really good! A few comments.

Generally, one other thing I've found – creating a rule without tags gives `Uncaught TypeError: Cannot read properties of undefined (reading 'toString')`.

File: apps/rule-manager/client/src/ts/components/LineBreak.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+export const LineBreak = () => <hr style={{
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply apps/rule-manager/client/src/ts/components/LineBreak.tsx]: I have now done so.

**Comment by rhystmills** [comment]
I believe I've sorted that tag problem, thanks for the review!

File: apps/rule-manager/client/src/ts/components/CategorySelector.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        if (!normalizedSearchValue) {
+            return;
+        }
+
+        const newOption = {
+            label: searchValue,
+        };
+
+        // Create the option if it doesn't exist.
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply apps/rule-manager/client/src/ts/components/CategorySelector.tsx]: I've set tags and categories to use a hardcoded list for now. The hardcoded tags are very inconsistent in terms of case etc but are what's in the sheet currently. We can revisit later after a tidy up!

File: apps/rule-manager/client/src/ts/components/TagsSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            return;
+        }
+
+        const newOption = {
+            label: searchValue,
+        };
+
+        // Create the option if it doesn't exist.
+        if (
+            flattenedOptions.findIndex(
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply apps/rule-manager/client/src/ts/components/TagsSelector.tsx]: (Now done)

File: apps/rule-manager/client/src/ts/components/RulesTable.tsx [hunk 4]
```diff
@@ -122,16 +126,25 @@ const RulesTable = () => {
...
+                        items={rules}
+                        columns={columns}
+                        pagination={true}
+                        sorting={sorting}
+                        search={search}
+                    />
+                }
+            </EuiFlexItem>
+            <EuiFlexItem grow={1}>
+                <RuleForm fetchRules={fetchRules}/>
```
Comment (jonathonherbert) [inline apps/rule-manager/client/src/ts/components/RulesTable.tsx:144]: Non-blocking – wonder if this is better expressed as `onCreate` or `onChange`, that is to say, our RuleForm shouldn't know about fetching rules, but should provide a hook to do something when users click 'create' (or, later on, 'update')

File: apps/rule-manager/client/src/ts/components/TagsSelector.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import {useEffect, useState} from "react";
+import {EuiFormRow, EuiComboBox} from "@elastic/eui";
+import React from "react";
+import { MetadataOption } from "./CategorySelector";
+import { PartiallyUpdateRuleData, RuleFormData } from "./RuleForm";
+
+const existingTags = [
```
Comment (jonathonherbert) [inline apps/rule-manager/client/src/ts/components/TagsSelector.tsx:7]: Non-blocking – should this live in 'constants'?

**Review by jonathonherbert — APPROVED** [review]
LGTM! A couple non-blocking comments.

File: apps/rule-manager/client/src/ts/components/TagsSelector.tsx (hunk 5 above)
Comment (rhystmills) [reply apps/rule-manager/client/src/ts/components/TagsSelector.tsx]: Good idea, now done!

File: apps/rule-manager/client/src/ts/components/RulesTable.tsx (hunk 4 above)
Comment (rhystmills) [reply apps/rule-manager/client/src/ts/components/RulesTable.tsx]: I've gone with `onRuleUpdate` for now as that's specifically when it's called by the RuleForm currently

**Review by jonathonherbert — APPROVED** [review]

**Pull request merged**